plugins/woocommerce-admin/client/products/layout/product-form-layout.tsx

```diff
@@ -0,0 +1,33 @@
+/**
+ * Internal dependencies
+ */
+import { ProductCategoryLayout } from './product-category-layout';
+import { ProductFieldLayout } from './product-field-layout';
+
+import './product-form-layout.scss';
+
+type ProductFormLayoutProps = {
+	categories: {
+		id: string;
+		title: string;
+		description: string;
+		fields: JSX.Element[];
+	}[];
+};
+export const ProductFormLayout: React.FC< ProductFormLayoutProps > = ( {
+	categories,
+} ) => {
+	return (
+		<div className="product-form-layout">
+			{ categories.map( ( { id, title, description, fields } ) => (
+				<div key={ id } className="product-form-layout__category">
+					<ProductCategoryLayout
+						title={ title }
+						description={ description }
+					/>
+					<ProductFieldLayout>{ fields }</ProductFieldLayout>
+				</div>
+			) ) }
```

Comment:
A slotfill could be added after this `categories` map, or have a slotfill after each section, that would be named something to the effect of `{section_name}_after`